src/app-layer-smtp.c

```diff
@@ -734,39 +567,37 @@ static int SMTPProcessCommandBDAT(
     SCReturnInt(0);
 }
 
-static void SetMimeEvents(SMTPState *state)
+static void SetMimeEvents(SMTPState *state, uint32_t events)
 {
-    if (state->curr_tx->mime_state->msg == NULL) {
+    if (events == 0) {
         return;
     }
 
-    /* Generate decoder events */
-    MimeDecEntity *msg = state->curr_tx->mime_state->msg;
-    if (msg->anomaly_flags & ANOM_INVALID_BASE64) {
+    if (events & MIME_ANOM_INVALID_BASE64) {
         SMTPSetEvent(state, SMTP_DECODER_EVENT_MIME_INVALID_BASE64);
     }
-    if (msg->anomaly_flags & ANOM_INVALID_QP) {
+    if (events & MIME_ANOM_INVALID_QP) {
         SMTPSetEvent(state, SMTP_DECODER_EVENT_MIME_INVALID_QP);
     }
-    if (msg->anomaly_flags & ANOM_LONG_LINE) {
+    if (events & MIME_ANOM_LONG_LINE) {
         SMTPSetEvent(state, SMTP_DECODER_EVENT_MIME_LONG_LINE);
     }
-    if (msg->anomaly_flags & ANOM_LONG_ENC_LINE) {
+    if (events & MIME_ANOM_LONG_ENC_LINE) {
         SMTPSetEvent(state, SMTP_DECODER_EVENT_MIME_LONG_ENC_LINE);
     }
-    if (msg->anomaly_flags & ANOM_LONG_HEADER_NAME) {
+    if (events & MIME_ANOM_LONG_HEADER_NAME) {
         SMTPSetEvent(state, SMTP_DECODER_EVENT_MIME_LONG_HEADER_NAME);
     }
-    if (msg->anomaly_flags & ANOM_LONG_HEADER_VALUE) {
+    if (events & MIME_ANOM_LONG_HEADER_VALUE) {
         SMTPSetEvent(state, SMTP_DECODER_EVENT_MIME_LONG_HEADER_VALUE);
     }
-    if (msg->anomaly_flags & ANOM_MALFORMED_MSG) {
+    if (events & MIME_ANOM_MALFORMED_MSG) {
         SMTPSetEvent(state, SMTP_DECODER_EVENT_MIME_MALFORMED_MSG);
     }
-    if (msg->anomaly_flags & ANOM_LONG_BOUNDARY) {
+    if (events & MIME_ANOM_LONG_BOUNDARY) {
         SMTPSetEvent(state, SMTP_DECODER_EVENT_MIME_BOUNDARY_TOO_LONG);
     }
-    if (msg->anomaly_flags & ANOM_LONG_FILENAME) {
+    if (events & MIME_ANOM_LONG_FILENAME) {
```

Comment:
Do we still need these ?

Reply:
why / why not?

Reply:
This zen answer tells me that you do not need these if you do not know if you need them ;-p

Reply:
Confusing. If we need them depends on where they are used, I think.

Reply:
They are used to set events : anomaly and rules with `app-layer-event`

Putting them